libs/contact/src/effects/contact.effects.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable, Inject } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { Action } from '@ngrx/store';
+import { Observable, of, EMPTY } from 'rxjs';
+import { switchMap, map, catchError } from 'rxjs/operators';
+
+import { DaffContactDriver } from '../driver/injection-tokens/contact-driver.token';
+import { DaffContactServiceInterface } from '../driver/interfaces/contact-service.interface';
+
+import { DaffContactActionTypes, DaffContactSubmit, DaffContactCancel, DaffContactSuccessSubmit, DaffContactFailedSubmit, DaffContactRetry } from '../actions/contact.actions';
+
+@Injectable()
+export class DaffContactEffects<T, V>{
+  constructor(
+    private actions$: Actions,
+    @Inject(DaffContactDriver) private driver: DaffContactServiceInterface<T, V>
+  ) { }
+
+  trySubmission$: Observable<Action> = createEffect(() => this.actions$.pipe(
+    ofType(DaffContactActionTypes.ContactSubmitAction,
+      DaffContactActionTypes.ContactRetryAction,
+      DaffContactActionTypes.ContactCancelAction),
+    switchMap((action: DaffContactSubmit<T> | DaffContactRetry<T> | DaffContactCancel) => {
+      if ((action instanceof DaffContactCancel)) {
+        return EMPTY;
+      }
+      else {
+        return this.submitContact(action, action.payload);
+      }
+    })
+  ));
+  private submitContact(action: Action, payload: any): Observable<Action> {
```

Comment:
doesn't look like the first `action` parameter is needed

Reply:
also, payload should be typed as `T`

Reply:
probably call the parameter contact instead of payload, since `submitContact` doesn't need to know it's getting an action.payload.

Reply:
Okay this should be taken care of in e775d4f9f31c6f40edbfe5252e1014a0e244e0c8